server/src/domains/products/products.service.ts

```diff
@@ -28,4 +41,108 @@ export class ProductsService {
       relations: { category: true, components: true },
     });
   }
+
+  private async createProductComponents(
+    componentIds: number[],
+  ): Promise<Component[]> {
+    const productComponents: Component[] = [];
+
+    for (const id of componentIds) {
+      const componentFromBd = await this.componentsService.getById(id);
+      if (componentFromBd) {
+        productComponents.push(componentFromBd);
+      }
+    }
+
+    return productComponents;
+  }
+
+  async create(
+    dto: CreateProductDto,
+    image: Express.Multer.File,
+  ): Promise<Product> {
+    const productCategory = await this.categoriesService.getById(
+      dto.categoryId,
+    );
+
+    if (!productCategory) {
+      throw new BadRequestException();
+    }
+
+    const fileName = this.filesService.createNewFile(
```

Comment:
Can this be encapsulated in the Image Service? Because the way we store images can change